UI/src/i18n.js

```diff
@@ -30,12 +30,25 @@ const i18n = createI18n({
 });
 
 export async function setI18nLanguage(lang) {
-    const locale = _mapLocale(lang.value);
+    let locale = _mapLocale(lang.value);
 
     // If the language hasn't been loaded yet
-    if (!i18n.global.availableLocales.includes(locale)) {  
-        const _messages = await import(/* webpackChunkName: "lang-[request]" */ `@/locales/${locale}.json`);
-        i18n.global.setLocaleMessage(locale, _messages.default);
+    if (!i18n.global.availableLocales.includes(locale)) {
+        try {
+            const _messages = await import(/* webpackChunkName: "lang-[request]" */ `@/locales/${locale}.json`);
+            i18n.global.setLocaleMessage(locale, _messages.default);
+        }
+        catch (e) {
+            const strippedLocale = locale.replace(/_[a-z]+/i, '');
+            try {
+                const _messages = await import(/* webpackChunkName: "lang-[request]" */ `@/locales/${strippedLocale}.json`);
+                i18n.global.setLocaleMessage(strippedLocale, _messages.default);
+                locale = strippedLocale;
+            }
+            catch (f) {
+                /* eslint-disable no-empty */
```

Comment:
you might want to set `locale = "en";` here otherwise it may be set to a non-existing locale `xy_AB` when the default locale is `en` when it is used later on at line 53. Not sure this is actually an issue in practice but then the `locale` variable should at least match the currently set locale.

Reply:
not sure about the `i18n` api but maybe you can get the current locale from there

Reply:
@ylavoie if we don't change locale to the fallback here, below, the page's text direction gets set to the direction associated with the locale passed in, but the actual locale is "en" so that it should be ltr, even if the preferred locale is arabic...

Reply:
ltr should always follow the effective locale.